movies-list

Adashevski opened the pull request:
Dodaje początkową listę filmów na podstawie trendów z ostatniego tygodnia.
Style zrobiłem "aby było"- skupiłem się na funkcjonalności. Od razu widzę, że zbyt długi tytuł filmu(gdy zajmuje dwie linijki) wypycha img w górę.

Adashevski committed:
movies-list

Adashevski committed:
add rating

arc4dev commented:
Super to wyglada! Jedna rzecz tylko zauwazylem co do architektury. Twoja funkcja getDataFromAPI renderuje filmy na podstawie URL, a default URL to sa filmy trending, wiec mozemy ja uzywac tak na prawde tez do szukania filmow po nazwie podajac odpowiedni URL, etc. Mogloby to wygladac w taki sposob: 
Plik index.js
`import './sass/main.scss';

import { renderMoviesList } from './js/movies-list';

// Render trending movies on load
renderMoviesList();`

Plik movies-list.js: 
`export const renderMoviesList = async (searchURL = TRENDING_URL) => {
  moviesContainer.innerHTML = '';
  // zmienne do importu
  const movieGenres = await getGenres(movieGenresURL);
  const tvGenres = await getGenres(TVGenresURL);
  const moviesList = getMovies(searchURL).then(response => {
    listBuilder(response);
  });
};`

Plik config.js: 
`export const API_KEY = 'dd28573eee75132a5c1e09d16e28974f';
export const API_URL = 'https://api.themoviedb.org/3/';
export const API_LANGUAGE = 'en-US';

// API URL's
export const TRENDING_URL = `${API_URL}trending/movie/week?api_key=${API_KEY}`;`

W tym podejsciu mozemy latwo uzywac twojej funkcji na URL wyszukiwania filmow po nazwie w przyszlosci :D 
Traktujemy index.js jako dyrygent orkiestry i importujemy wszystkie funkcje do niego. To powinno zwiekszyc przejrzystosc i funkcjonalnosc naszego kodu 👍 Daj znac co myslisz.

Adashevski commented:
Dzięki! Dokładnie taki miałem zamysł. Tylko nie bardzo wiedziałem jak ugryźć kwestie exportów/importów. Stąd też te zmienne, które dają amunicje do dalszych działań:
const movieGenres
  const tvGenres 
  const moviesList

arc4dev commented:
A ! i jeszcze plik ze stylami w folderze musi sie znajdowac sass/partials, utils sluzy do roznych zasobow, zmiennych i tak dalej 😀

Adashevski committed:
update

-zmiana nazwy funkcji(renderMoviesList)
-export funkcji i wywołanie w pliku index.js
-utworzenie folderu sass/partials
-inne zmiany kosmetyczne

areksroczyk committed:
conflict

areksroczyk committed:
Merge branch 'feature/movies-list' of https://github.com/arc4dev/Projekt-JS-Filmoteka into feature/movies-list
